### PR TITLE
[quant][pt2] Handle constant conv args in prepare QAT fusion

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100941
* #100610
* __->__ #100525

Summary: Previously, we would only match and replace conv + BN
patterns with default constant args for conv (stride, padding,
dilation etc.). If the user sets one of these args to values
that are different from the default, we would simply not fuse
the pattern. This is due to a limitation in the subgraph
rewriter: see https://github.com/pytorch/pytorch/issues/100419.

This commit works around the above limitation by first
configuring the subgraph rewriter to ignore literals when
matching, and then manually copy over the constant args to the
new subgraph after `replace_pattern`.

Test Plan:
python test/test_quantization.py TestQuantizePT2E.test_prepare_qat_conv_bn_fusion_constant_args

Reviewers: jerryzh168, kimishpatel

Differential Revision: [D45515437](https://our.internmc.facebook.com/intern/diff/D45515437)

### CODE REVIEW

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

File: torch/ao/quantization/_pt2e/qat_utils.py [hunk 1]
```diff
@@ -119,9 +129,12 @@ def _fuse_conv_bn_qat(m: GraphModule) -> GraphModule:
             n = n.args[0]
 
         # Copy over metadata for all three nodes in [conv - bn - getitem]
+        # Also copy over constant args for conv
         for match_pattern_node, original_node in mr.nodes_map.items():
             if original_node.target == torch.ops.aten.convolution.default:
                 replacement_conv_node.meta = original_node.meta
+                # x, weight, bias, [stride, padding, dilation, transposed, output_padding, groups]
+                replacement_conv_node.args = replacement_conv_node.args[:3] + original_node.args[3:]
```
Comment (jerryzh168) [inline torch/ao/quantization/_pt2e/qat_utils.py:137]: does this work? I remember when I'm doing similar things the original_node is actually a node in the updated graph as well so the information has been lost

Comment (andrewor14) [reply torch/ao/quantization/_pt2e/qat_utils.py:137]: Yes I ran into that before as well. The difference here is these args are literal constant args so they're inlined. You wouldn't be able to access node args from the conv node, however, e.g. the original conv weight or bias.

**Review by jerryzh168 — APPROVED** [review]

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45515437).

**Pull request closed without merging**